projects/plugins/boost/app/modules/optimizations/page-cache/pre-wordpress/class-filesystem-utils.php

```diff
@@ -45,6 +45,8 @@ public static function iterate_directory( $path, Path_Action $action ) {
 			$count += $action->apply_to_path( $file );
 		}
 
+		$count += $action->apply_to_path( new SplFileInfo( $path ) );
```

Comment:
I tried this and found it may produce errors in GC due to the directory not existing. I think we can either leave the directory alone or check for directory exists in the delete action.